motivaa-backend/motivaa-app/src/test/java/com/motivaa/control/HabitCreationServiceTest.java

```diff
@@ -0,0 +1,263 @@
+import com.motivaa.control.error_handling.exceptions.NotFoundException;
+import com.motivaa.control.error_handling.exceptions.RepositoryException;
+import com.motivaa.control.repository.MotivaaRepository;
+import com.motivaa.control.utility.MessageBundle;
+import static com.motivaa.control.utility.MessageBundle.INVALID_RECURRING_TYPE_ERROR_MESSAGE;
+import com.motivaa.entity.DaySpecificHabit;
+import com.motivaa.entity.Habit;
+import com.motivaa.entity.NotDaySpecificHabit;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import static org.mockito.ArgumentMatchers.any;
+import org.mockito.Mockito;
+import static org.mockito.Mockito.doNothing;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.Arrays;
+
+@SpringBootTest
+@ActiveProfiles("test")
+public class HabitCreationServiceTest {
+
+    @MockBean
+    private MotivaaRepository motivaaRepository;
+    @MockBean
+    private CommonHabitServices commonHabitServices;
+
+    @Autowired
+    HabitCreationService habitCreationService;
+
+    @Nested
+    class Success_Scenarios {
+
+        @Test
+        void valid_daySpecificHabit_should_be_successful() throws Exception {
+
+            doNothingMockOfValidateUsers(HabitRequestCreator.VALID_USER_UUID);
+
+            doNothing().when(commonHabitServices).validateRecurringFieldsForType(
+                            HabitRequestCreator.VALID_RECURRING_TYPE_SPECIFIC_DAY,
+                            HabitRequestCreator.VALID_LIST_OF_RECURRING_DAYS,
+                            null);
+
+            doNothing().when(motivaaRepository).saveHabit(new DaySpecificHabit(
+                    HabitRequestCreator.VALID_USER_UUID,
+                    HabitRequestCreator.VALID_NAME,
+                    HabitRequestCreator.VALID_RECURRING_TYPE_SPECIFIC_DAY,
+                    Arrays.asList(HabitRequestCreator.VALID_LIST_OF_RECURRING_DAYS.split(";")),
+                    Integer.parseInt(HabitRequestCreator.VALID_PRIORITY),
+                    HabitRequestCreator.VALID_COLOR));
+
+            Habit habit = habitCreationService.createHabit(
+                    HabitRequestCreator.VALID_USER_UUID,
+                    HabitRequestCreator.VALID_NAME,
+                    HabitRequestCreator.VALID_RECURRING_TYPE_SPECIFIC_DAY,
+                    HabitRequestCreator.VALID_LIST_OF_RECURRING_DAYS,
+                    null,
+                    Integer.parseInt(HabitRequestCreator.VALID_PRIORITY),
+                    HabitRequestCreator.VALID_COLOR);
+
+            assertEquals(
+                    DaySpecificHabit.class,
+                    habit.getClass());
+            assertEquals(
```

Comment:
Maybe could create a method in TestUtils to assert the Habit's common parts, so you would only assert here what is specific, so there wouldn't be this many code duplication here.

Reply:
Or use what I suggested above if this test is only about testing the validation and not about the values.